File: challenges/encryption/problem.rb
#!/bin/ruby

def encryption(s)
    sqr = Math.sqrt s.length
    grid = s.scan(/\w{0,#{sqr.ceil}}/).map { |m| m.split('') }
    out = []
    (0...sqr.ceil).each do |n|
        (0...sqr.round).each { |nn| out.push grid[nn][n] if grid[nn][n] && grid[nn][n] != "" }
        out.push ' '
    end
    return out.join.strip
end

s = gets.strip
result = encryption(s.gsub(/\s+/, ""))
puts result

File: challenges/time-conversion/problem.py
#!/bin/python3

import sys

def timeConversion(s):
    ampm = s[-2:]
    t = s[:-2].split(':')
    if t[0] == '12': t[0] = '00'
    if ampm == 'PM': t[0] = str(int(t[0]) + 12)
    return ':'.join(t)

s = input().strip()
result = timeConversion(s)
print(result)
